userspace/falco/app_actions/init_falco_engine.cpp

```diff
@@ -0,0 +1,111 @@
+/*
+Copyright (C) 2022 The Falco Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "defined_app_actions.h"
+
+namespace falco {
+namespace app {
+
+void configure_output_format(application &app)
+{
+	std::string output_format;
+	bool replace_container_info = false;
+
+	if(app.options().print_additional == "c" || app.options().print_additional == "container")
+	{
+		output_format = "container=%container.name (id=%container.id)";
+		replace_container_info = true;
+	}
+	else if(app.options().print_additional == "k" || app.options().print_additional == "kubernetes")
+	{
+		output_format = "k8s.ns=%k8s.ns.name k8s.pod=%k8s.pod.name container=%container.id";
+		replace_container_info = true;
+	}
+	else if(app.options().print_additional == "m" || app.options().print_additional == "mesos")
+	{
+		output_format = "task=%mesos.task.name container=%container.id";
+		replace_container_info = true;
+	}
+	else if(!app.options().print_additional.empty())
+	{
+		output_format = app.options().print_additional;
+		replace_container_info = false;
+	}
+
+	if(!output_format.empty())
+	{
+		app.state().engine->set_extra(output_format, replace_container_info);
+	}
+}
+
+static run_result run_init_falco_engine(base_action &act)
+{
+	run_result ret = {true, "", true};
+	application &app = ((action &) act).app;
```

Comment:
For the purposes of testing, I needed a way to set up a bunch of actions and make tests for them, without bringing in all of the objects in application::action_state(). Since the catch2 program just grabs individual files as needed, and not the whole falco application code, there was a cascade of dependencies. The base_action class was an action without any explicit application, but the function signatures wouldn't work with the derived objects.

Moot now, as the actions are gone.